### PR TITLE
DOC: Tighten howto-docs guide #16259

### PR DESCRIPTION
**Pull request by bjnath**
PR picks up from #16259. It addresses the two comments made in the earlier PR.

image

### CODE REVIEW

**Comment by bjnath** [comment]
Is anything further standing in the way of merging this PR? Happy to address any remaining feedback.

The [most recent review comment](https://github.com/numpy/numpy/pull/16259#pullrequestreview-417855404) ("Overall, I think this looks good!") suggested the PR might be ready to merge once two points of feedback were addressed.  I've made those changes.

Please let me know if anything from me is still needed.

**Review by rossbar — APPROVED** [review]
LGTM - AFAICT the content is largely consistent between this version and the original and it looks like all of the major points are covered.

On a separate (but related) note: at one point in the discussion, there was talk of perhaps coming up with a companion doc (either a "how-to" or a "tutorial") that explicitly walks users through the process of building docs. I think that's a great idea as the doc building process can be a bit involved, especially for new users.

**Comment by bjnath** [comment]
8192a03 addresses the comments about "handwritten" by @rossbar and @mattip in https://github.com/numpy/numpy/pull/16385/#discussion_r438320504 and has
other revisions throughout.

<details><summary>New page:</summary>

image

</details>

File: doc/source/dev/howto-docs.rst [hunk 1]
```diff
@@ -4,142 +4,154 @@
...
+<https://mail.python.org/mailman/listinfo/numpy-discussion>`__. Minutes are
+taken `on hackmd.io <https://hackmd.io/oB_boakvRqKR-_2jRV-Qjg>`__ and stored
 in the `NumPy Archive repository <https://github.com/numpy/archive>`__.
 
+*************************
+Overview
+*************************
+:ref:`NumPy Documentation <numpy_docs_mainpage>` falls into two categories:
+
+- **Hand-written:** Tutorials, How-Tos, Explanations. More on these categories below.
```
[the remark this answers is not in the document]
Comment (bjnath) [reply doc/source/dev/howto-docs.rst]: Please see if the wording in 8192a03 is an improvement.

**Comment by bjnath** [comment]
2c145be fixes a repeated word.

File: doc/source/dev/howto-docs.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (bjnath) [reply doc/source/dev/howto-docs.rst]: 6043b0f revises it again to make it more straightforward.

**Comment by bjnath** [comment]
6043b0f has more straightforward wording in the section about generated docs and adds verbs to two headings.

<details><summary>New page:</summary>

image


</details>

**Comment by bjnath** [comment]
75c5908 fixes another repeated word.

<details><summary>New page:</summary>
image

</details>

**Comment by mattip** [comment]
@bjnath you do not need to attach an image. If you click on the "Details" link in the "ci/circleci: build artifact" build, you will be directed to the built documentation, and you can paste a link to the page here instead.

Also, you don't need to comment on each commit (unless you want to). Those of us who are subscribed to the PR get a mail each time you commit with the commit message. That is why we request that the commit messages be [formatted](https://13992-908607-gh.circle-artifacts.com/0/doc/build/html/dev/development_workflow.html#writing-the-commit-message) in a specific way: that make filtering the many emails easier.

**Comment by mattip** [comment]
Thanks @bjnath

**Comment by bjnath** [comment]
@mattip -- Thank you for the suggestions. 

> you don't need to comment on each commit

In other words, the commit message is sufficient and I don't have to return to this page and summarize it?

> commit messages be formatted in a specific way

I'd begun all the commits with DOC and referenced the PR in the first line -- is there something else stylistically I should do when the commit updates a pending PR?

> do not need to attach an image

Great! I'll attach the link as you suggest.


Sorry for the last-minute flurry of changes -- I appreciate your patience.

**Comment by mattip** [comment]
> the commit message is sufficient

As a rule of thumb: yes. If the new commit is missed you could ping a reviewer after a day or two (a week for really complicated things).

>I'd begun all the commits with DOC and referenced the PR in the first line -- is there something else stylistically I should do when the commit updates a pending PR?

No, my bad, sorry. I wonder why my filter failed?

Thanks for the contributions.

**Pull request merged**